packages/ngx-schematics/src/page/index.ts

```diff
@@ -0,0 +1,193 @@
+          /* webpackExports: "Page${strings.classify(options.name)}RoutingModule" */
+          '${strings.dasherize(options.project)}').then(m => m.Page${strings.classify(
+      options.name
+    )}RoutingModule),
+      data: {
+        title: $localize\`:@@${strings.dasherize(options.project)}/page-${strings.dasherize(
+      options.name
+    )}#TITLE:${strings.nameify(options.name)}\`,
+        iconClass: '${options.iconClass}',
+        requiredFunctionalities: '${options.pageId}'
+      }
+  }`;
+
+    insertRoutesMetadata(source, {}, toInsert);
+    tree.overwrite(metadataPath, source.getFullText());
+  };
+}
+
+function getEntrypointMetadata(options: Schema): UpdateMetadataOptions {
+  if (options.entrypoint === 'Menu Item') {
+    return {
+      identifier: MetadataProperty.MenuItem,
+      imports: {},
+      toInsert: `\
+      {
+        id: '${options.pageId}',
+        menuGroupId: '${options.menuGroupId}',\
+        ${
+          options.menuSubGroupId!.length > 0 ? `\nmenuSubGroupId: '${options.menuSubGroupId}',` : ''
+        }
+        title: $localize\`:@@${strings.dasherize(options.project)}/page-${strings.dasherize(
+        options.name
+      )}#TITLE:${strings.nameify(options.name)}\`,
+        actionId: '${options.pageId}',
+        position: 1,
+        iconClass: '${options.iconClass}',
+        requiredFunctionalities: '${options.pageId}'
+      }`
+    };
+  }
+
+  return {
+    identifier: MetadataProperty.ActionButton,
+    imports: {},
+    toInsert: `\
+    {
+      id: '${options.pageId}',
+      actionId: '${options.pageId}',
+      title: $localize\`:@@${strings.dasherize(options.project)}/page-${strings.dasherize(
+      options.name
+    )}#TITLE:${strings.nameify(options.name)}\`,
+      iconClass: '${options.iconClass}',
+    }`
+  };
+}
+
+export default function (_options: Schema): Rule {
+  return async (tree: Tree, _context: SchematicContext) => {
+    const workspace = await readWorkspace(tree);
+    const project = workspace.projects.get(_options.project);
+
+    if (!project) {
+      throw new SchematicsException(`Project "${_options.project}" does not exist.`);
+    }
+
+    if (_options.path === undefined) {
+      _options.path = getDefaultPath(project);
+    }
+
+    if (!_options.name) {
+      throw new SchematicsException(`Action name is required`);
+    }
+
+    if (!_options.pageId) {
+      throw new SchematicsException(`Page Id is required`);
+    }
+
+    if (!_options.iconClass) {
+      throw new SchematicsException(`Icon class is required`);
+    }
+
+    if (_options.entrypoint === 'Menu Item' && !_options.menuGroupId) {
+      const questions: inquirer.InputQuestion[] = [
+        {
+          type: 'input',
+          name: 'menuGroupId',
+          message: 'What is the Menu Group Id?'
+        },
+        {
+          type: 'input',
+          name: 'menuSubGroupId',
+          message: 'What is the Menu Sub Group Id? (ignore if not applicable)'
```

Comment:
"(ignore if not applicable)"
I don't think you need part in the question but I leave it to you.